lib/descriptor/type/raw.js

```diff
@@ -0,0 +1,78 @@
+/*!
+ * raw.js - raw descriptor object for bcoin
+ * Copyright (c) 2023, the bcoin developers (MIT License).
+ * https://github.com/bcoin-org/bcoin
+ */
+
+'use strict';
+
+const Descriptor = require('../descriptor');
+const common = require('../common');
+const {RAW} = common.types;
+const {isType, strip, checkChecksum} = common;
+const {TOP} = common.scriptContext;
+const base16 = require('bcrypto/lib/encoding/base16');
+const Script = require('../../script/script');
+const Network = require('../../protocol/network');
+const assert = require('bsert');
+
+/**
+ * Raw Descriptor
+ * Parsed raw descriptor object
+ * @property {String} name
+ * @property {Script} script
+ * @property {Network} network
+ * @extends Descriptor
+ */
+
+class RawDescriptor extends Descriptor {
+  constructor(options) {
+    super();
+    this.name = RAW;
+
+    if (options) {
+      this.fromOptions(options);
+    }
+  }
+
+  fromOptions(options) {
+    assert(options.script, 'Must pass script');
+    this.network = Network.get(options.network);
+    this.script = options.script;
+    return this;
+  }
+
+  static fromOptions(options) {
+    return new this().fromOptions(options);
+  }
+
+  fromString(str, network, context) {
+    str = checkChecksum(str);
+
+    assert(isType(RAW, str), 'Invalid raw descriptor');
+    assert(context === TOP, 'Can only have raw() at top level');
+
+    str = strip(RAW, str);
+    assert(base16.test(str), 'Raw script is not hex'); // check hex format
+    const script = Script.fromRaw(str, 'hex');
```

Comment:
I ran the raw descriptor test `input : 'raw(asdf)'` without this check and surprisingly the string was parsed without throwing any error (It should throw error since argument inside raw is not hex string). Although the script field was empty so I think we can have this check.